Readme content for Domain Explorer (#180)
kiegroup/kogito-apps: management-console/README.md

The native build of the application bundling in the React JS frontend does not differ from the instructions above. The only thing that's new is again the invocation of UI specific profiles.
```
mvn package -Dui -Dnative
```

## Domain Explorer

> Domain Explorer allows you to check domain related data and process instances based on domains that are available.

Reviewer: ```suggestion
Domain Explorer allows you to inspect domain related data and process instances based on domains that are available.
```

Author: @cristianonicolai done

You can navigate to Domain Explorer by clicking "**Domain Explorer**" tab available on the sidebar, which will take you landing page of
domain explorer and there you can see a list of available Domains.

> <img src="docs/DELandingpage.png" height="400px" width="700px">

Reviewer: image seems outdated, can you please update with the latest changes from Sara?

Author: done

> Selecting a domain on the landing page will direct you to a dashboard, where you can see a table with domain data and list of processes associated to it.

Reviewer: ```suggestion
Selecting a domain on the landing page will direct you to a dashboard, where you can view all domain related data and list of processes associated with it.
```

Author: @cristianonicolai done

> Selecting a domain on the landing page will direct you to a dashboard, where you can view all domain related data and list of processes associated with it.

Reviewer 2: @Sara4994 makes sense talk about 'process instances' instead of 'processes'?

Author: done

### Column Picker

> Domain Explorer allows you to select the domain attributes that you wish to see on the table.

Reviewer: ```suggestion
Domain Explorer allows you to select the domain attributes that you wish to see on the result table.
```

Author: done

> This column picker is a multi select dropdown contains domain related attributes, that allows you to select any number of options available in the dropdown, there are few options that are selected by default which can be changed of your choice and the options available in the dropdown follows certain nested pattern. 

Reviewer: ```suggestion
This column picker is a multi-select dropdown that contains the available domain attributes, that allows you to select any number of options available in the dropdown, there are few options that are selected by default which can be changed of your choice. The options available in the dropdown follows a nested pattern, displayed the entire graph of attributes available in the specific domain.
```

Author: done

eg: "**hotel/address**" denotes address attribute of hotel.

> After selecting/deselecting options from the dropdown, clicking on "**Apply columns**" would get you domain data on the table.

Reviewer 3: After selecting/deselecting options from the dropdown, clicking on "**Apply columns**" would populate domain data on the table. - is this usage fine?

Author: @AjayJagan  yep, done

![Column Picker](./docs/pickColumns.gif "Column Picker")

### Refresh option

> There can be a possibilites of,

Reviewer 3: `There can be possibilities of : `  - is this ok to add?

Author: yep. it was a typo. removed "a"

* change in state of processes, while performing operations like skip, retry or abort
* New process instance being added.

> In order to check the current state of processes or current list of processes available on the domain you can always do a refresh of the table by clicking icon placed next to Apply columns button. This will get you current and updated content to the table.

Reviewer 2: @Sara4994 to be consistent here I would add formatting to the reference to the button: "**Apply columns**"

Author: @nmirasch done

### Dashboard

> The table on the Domain Explorer will provide you information about Domain Data and list of process associated with it.

Reviewer: ```suggestion
The table on the Domain Explorer will provide you information about the specific domain data and list of processes associated with it.
```

Author: done

> The table on the Domain Explorer will provide you information about the specific domain data and list of processes associated with it.

Reviewer 2: @Sara4994  maybe we can talk here about 'process instances' instead of 'processes' to make explicit the link with the other section.

Author: done

To check the list of processes, you may have to expand each row of the table by clicking on the **carret** icon available on each row.

<img src="docs/expandRow.gif" height="400px" width="700px">

### Navigate to process details

> To get to know more about process, you can always click on any process instance to direct to process details page and there you can see details about the process, timeline and process variables.

Reviewer: ```suggestion
To get to know more about a process, you can always click on any process instance to direct to process details page and there you can see details about the process as well as the timeline and process variables.
```

Author: done

> To get to know more about a process, you can always click on any process instance to direct to process details page and there you can see details about the process as well as the timeline and process variables.

Reviewer 3: I think we can add 'To know more about a process', wdyt @Sara4994

Author: done

> To get to know more about a process, you can always click on any process instance to direct to process details page and there you can see details about the process as well as the timeline and process variables.

Author: ```suggestion
To get to know more about a process, you can always click on any process instance to direct to process details page and there you can see details about the process as well as the timeline and process variables
```
